### ISSUE TITLE
Field.getDataType().defaultValue() value is 'null' instead of actual value.

### ISSUE THREAD

**Issue by rbuehlma** [jOOQ/jOOQ#8875]
### Expected behavior and actual behavior:
The default value of a column returned by the meta() is 'null' (String, not the null value) instead of the actual default if the default uses a function.

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

```
     private static final String CREATE_TABLE_FUNCTION_DEFAULT_SQL =
        "create table \"test-target\".test_table (" +
            "id bigint default char_length('defaultValue'::text) " +
            ")";

    @Test
    public void jooqDefaultTest() {
        DSLContext context = DSL.using("jdbc:postgresql://localhost:5432/test?user=test&password=test");
        List<org.jooq.Schema> schemas = context.meta().getSchemas("test-target");
        Meta meta = context.meta(schemas.get(0));
        context.execute(CREATE_TABLE_FUNCTION_DEFAULT_SQL);
        Table<?> table = meta.getTables("test_table").get(0);
        Field<?> defaultValue = table.field("id").getDataType().defaultValue();
        assertThat(defaultValue.getName(), is("char_length('defaultValue'::text)"));
    }
```
This sample does work if the default is something simple like e.g. 1

### Versions:

- jOOQ: 3.11.11
- Java: 1.8
- Database (include vendor): PostgreSQL 11.4
- OS: Ubuntu 19.04
- JDBC Driver (include name if inofficial driver): postgresql 42.2.5

**Comment by lukaseder** [comment]
Thank you very much for your report. There are a few known issues around this `defaultValue()` functionality, both at runtime (e.g. #8469), and in the code generator (e.g. #7849). We'll address them thoroughly in jOOQ 3.13, and make sure, the issue you reported will be covered as well.

**Comment by lukaseder** [comment]
https://github.com/jOOQ/jOOQ/issues/8469 is now implemented for jOOQ 3.15. We'll be adding some extensive integration tests to make sure these default expressions are parsed as much as possible, and reproduced in the meta model.

**Comment by lukaseder** [comment]
I can confirm that with the current version 3.15.0-SNAPSHOT, the following code:

```java
try {
    List<org.jooq.Schema> schemas = ctx.meta().getSchemas("public");
    Meta meta = ctx.meta(schemas.get(0));
    ctx.execute(CREATE_TABLE_FUNCTION_DEFAULT_SQL);
    Table<?> table = meta.getTables("test_table").get(0);
    Field<?> defaultValue = table.field("id").getDataType().defaultValue();
    System.out.println(ctx.render(defaultValue));
}
finally {
    ctx.execute("drop table if exists test_table");
}
```

prints:

```
char_length(cast('defaultValue' as text))
```
